doc/source/whatsnew/v0.20.0.txt

```diff
@@ -382,6 +383,18 @@ Other Enhancements
 Backwards incompatible API changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+Deprecate Automatic Zero Filling of Missing Values in pd.get_dummies
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``pd.get_dummies`` function currently fills missing values with zero by default. This behavior is in conflict with the rest of the pandas API since missing values should be filled with ``fillna`` and missing values should be propagated through pandas transformations. In the future, ``get_dummies`` will propagate missing values by default. The recommended way to reproduce the current behavior of filling with zeros with the new, upcoming API is
+
+.. ipython: python
+
+  df = pd.get_dummies(df).fillna(0)
+
+For now, the current behavior of filling zeros by default has been kept, but not specifying a fill value with the ``fill_value`` keyword will raise a ``DeprecationWarning`` with the example above.
```

Comment:
yes specifying fill_value is better (otherwise u would end up with int64); though we do have downcast option in fillna (it's just not that aggressive)